lua/entities/gmod_wire_expression2/init.lua

```diff
@@ -142,8 +142,14 @@ function ENT:Execute()
 
 	for k, v in pairs(self.inports[3]) do
 		if self.GlobalScope[k] then
-			if wire_expression_types[self.Inputs[k].Type][3] then
-				self.GlobalScope[k] = wire_expression_types[self.Inputs[k].Type][3](self.context, self.Inputs[k].Value)
+			local CopyFunction = wire_expression_types[self.Inputs[k].Type][3]
+			if CopyFunction and not self.rawinputs[k] then
+				local Copy = CopyFunction(self.context, self.Inputs[k].Value)
+				if Copy then
```

Comment:
This test shouldn't happen. I think this will break boolean values, as a `false` will be turned into `nil`. It should always assign `Copy` regardless.